nixos/modules/services/web-apps/matomo.nix

```diff
@@ -127,27 +139,33 @@ in {
       # e.g. after restoring from backup or moving from another system.
       # Note that ${dataDir}/config/config.ini.php might contain the MySQL password.
       preStart = ''
+        # migrate data from piwik to matomo folder
+        if [ -d ${deprecatedDataDir} ]; then
+          echo "Migrating from ${deprecatedDataDir} to ${dataDir}"
+          mv -T ${deprecatedDataDir} ${dataDir}
+          rmdir ${deprecatedDataDir}
+        fi
```

Comment:
Huh.. this shouldn't have failed on me. 

Systemd output:
``` log
warning: the following units failed: matomo_setup_update.service

● matomo_setup_update.service
   Loaded: loaded (/nix/store/9f2hkvllc46vjp3x0h92vfnwljsf381g-unit-matomo_setup_update.service/matomo_setup_update.service; enabled; vendor preset: enabled)
   Active: failed (Result: exit-code) since Wed 2018-01-31 22:59:45 EST; 16s ago
  Process: 29211 ExecStartPre=/nix/store/wmg25rziv6zcm0hcr0q4wd1dzjqw7scj-unit-script/bin/matomo_setup_update-pre-start (code=exited, status=1/FAILURE)

Jan 31 22:59:45 jarmac.org systemd[1]: Starting matomo_setup_update.service...
Jan 31 22:59:45 jarmac.org matomo_setup_update-pre-start[29211]: Migrating from /var/lib/piwik to /var/lib/matomo
Jan 31 22:59:45 jarmac.org matomo_setup_update-pre-start[29211]: rmdir: failed to remove '/var/lib/piwik': No such file or directory
Jan 31 22:59:45 jarmac.org systemd[1]: matomo_setup_update.service: Control process exited, code=exited status=1
Jan 31 22:59:45 jarmac.org systemd[1]: Failed to start matomo_setup_update.service.
Jan 31 22:59:45 jarmac.org systemd[1]: matomo_setup_update.service: Unit entered failed state.
Jan 31 22:59:45 jarmac.org systemd[1]: matomo_setup_update.service: Failed with result 'exit-code'.
warning: error(s) occurred while switching to the new configuration
```
I personally would not worry about this as it went away on retry, and there's no reason to suspect a race condition here.

Reply:
Tested it again, the `rmdir` is superfluous I think. Probably it's a leftover / merge error from before I discovered `mv -T`. Removed it now.